docs/tutorials/1_dqn_tutorial.ipynb

```diff
@@ -1,1144 +1,1156 @@
+    "\n",
+    "First, create a function to embed videos in the notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "ULaGr8pvOKbl"
+   },
+   "outputs": [],
+   "source": [
+    "def embed_mp4(filename):\n",
+    "  \"\"\"Embeds an mp4 file in the notebook.\"\"\"\n",
+    "  video = open(filename,'rb').read()\n",
+    "  b64 = base64.b64encode(video)\n",
+    "  tag = '''\n",
+    "  <video width=\"640\" height=\"480\" controls>\n",
+    "    <source src=\"data:video/mp4;base64,{0}\" type=\"video/mp4\">\n",
+    "  Your browser does not support the video tag.\n",
+    "  </video>'''.format(b64.decode())\n",
+    "\n",
+    "  return IPython.display.HTML(tag)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "9c_PH-pX4Pr5"
+   },
+   "source": [
+    "Now iterate through a few episodes of the Cartpole game with the agent. The underlying Python environment (the one \"inside\" the TensorFlow environment wrapper) provides a `render()` method, which outputs an image of the environment state. These can be collected into a video."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "owOVWB158NlF"
+   },
+   "outputs": [],
+   "source": [
+    "def create_policy_eval_video(policy, filename, num_episodes=5, fps=30):\n",
+    "  filename = filename + \".mp4\"\n",
+    "  with imageio.get_writer(filename, fps=fps) as video:\n",
+    "    for _ in range(num_episodes):\n",
+    "      time_step = eval_env.reset()\n",
+    "      video.append_data(eval_py_env.render())\n",
+    "      while not time_step.is_last():\n",
+    "        action_step = policy.action(time_step)\n",
+    "        time_step = eval_env.step(action_step.action)\n",
+    "        video.append_data(eval_py_env.render())\n",
+    "  return embed_mp4(filename)\n",
+    "\n",
+    "\n",
+    "\n",
+    "\n",
+    "create_policy_eval_video(agent.policy, \"trained-agent\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "povaAOcZygLw"
+   },
+   "source": [
+    "For fun, compare the trained agent (above) to an agent moving randomly. (It does not do as well.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "pJZIdC37yNH4"
+   },
+   "outputs": [],
+   "source": [
+    "create_policy_eval_video(random_policy, \"random-agent\")"
+   ]
+  }
+ ],
+ "metadata": {
+  "colab": {
+   "collapsed_sections": [],
+   "name": "DQN Tutorial.ipynb",
+   "private_outputs": true,
+   "provenance": [],
+   "toc_visible": true
+  },
+  "kernelspec": {
+   "display_name": "Python 3",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
```

Comment:
If you can remove from here down, through I think line 1142 I can just accept this with zero effort.  If not, I need to lookup if we really want this extra info.  Off the top of my head I don't know and it will take time to sort it out and see if it matches the other files and what to do. I likely to be overly transparent. I just don't know the possibly ramification of these additions.  Thank you for understanding and for the link fix.